boltons/dictutils.py

```diff
@@ -964,6 +964,49 @@ def subdict(d, keep=None, drop=None):
 
     keys = set(keep) - set(drop)
 
-    return dict([(k, v) for k, v in d.items() if k in keys])
+    return type(d)([(k, v) for k, v in d.items() if k in keys])
+
+
+class FrozenDict(dict):
+    _hash = None
```

Comment:
love the brevity -- could use `__slots__ = ('_hash',)` I think

Reply:
done!